src/iter.rs

```diff
@@ -150,30 +230,78 @@ impl<T: DoubleEndedIterator> DoubleEndedIterator for ProgressBarIter<T> {
     }
 }
 
+impl<T: ExactSizeIterator> ExactSizeIterator for ProgressBarIter<T> {
+    fn len(&self) -> usize {
+        self.it.len()
+    }
+}
+
 impl<T: FusedIterator> FusedIterator for ProgressBarIter<T> {}
 
+impl<S, T: Iterator<Item = S>> ProgressIterator for T {
+    fn progress_with(self, progress: ProgressBar) -> ProgressBarIter<Self> {
+        ProgressBarIter {
+            it: self,
+            progress,
+            hold_max: RingBufWrap::new(),
+        }
+    }
+}
+
+#[cfg(feature = "futures")]
+#[cfg_attr(docsrs, doc(cfg(feature = "futures")))]
+impl<S: futures_core::Stream + Unpin> futures_core::Stream for ProgressBarIter<S> {
+    type Item = S::Item;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let this = self.get_mut();
+        let item = std::pin::Pin::new(&mut this.it).poll_next(cx);
+        match &item {
+            std::task::Poll::Ready(Some(_)) => this.progress.inc(1),
+            std::task::Poll::Ready(None) => this.progress.finish_using_style(),
+            std::task::Poll::Pending => {}
```

Comment:
Import all of the types (`Pin`, `Context`, `Poll`) to avoid the unnecessary clutter from import paths.

Reply:
un-moved pre-existing code so this does not show up in the diff anymore